bip-tx-terminology.mediawiki

```diff
@@ -0,0 +1,286 @@
+We noticed that terminology to refer to the components of transactions and functions that scripts fulfill in transactions includes a broad spectrum of alternative jargon.
+In this document, we describe a set of terms that can be used to refer to transaction components by their function or their context.
+The terms are not entirely consistent in themselves since there are many terms that are already firmly established.
+Instead of striving for consistency, we try to relate terms to each other, explore the origins of terms, and fill in blanks.
+
+===Objects and their Serialization===
+
+We treat the objects of the Bitcoin Protocol (e.g. Transactions, Blocks) as distinct from their serialization.
+We distinguish concrete properties of transactions such as fields and scripts (labeled [TC] for Transaction Component) from artifacts of the transaction serialization (labeled [SA] for Serialization Artifact).
+We also introduce some concepts and ideas that are useful to describe aspects of transcations (labeled [AT] for Abstract Term).
+For example, version and locktime are fields that can be set during transaction creation and therefore properties of a transaction, while the input counter is a serialization artifact resulting from how the input list of the transaction is represented.
+Likewise, the _input script_ is a transaction field, but the length indicator of the input script a serialization artifact.
+
+===Definition of Terms===
+
+; Annex
+: [TC] A Witness Item reserved for future extensions, currently unused (see [[https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki#cite_note-5|BIP341]]).
+
+; Amount
+: [TC] The count of satoshis assigned to an output.
+
+; Bare Output Script
+: [AT] A Condition Script that is directly defined in the Output Script field. Used to specify all types of Bare Outputs including Pay to Pubkey, Pay to Public Key Hash, Pay to Multisig outputs, and arbitrary scripts.
+
+; Condition Script
+: [AT] Collective term for scripts that are not forwarding scripts and encode a set of conditions that could be fulfilled to spend a UTXO.  A scriptPubkey can either contain directly a condition script in the case of a bare output, or will commit to one or multiple condition scripts for other output types.  For P2TR the tweaked public key is considered a condition script as well as each Leaf Script in the script tree.  At the time of writing, there are four distinct Condition Scripts: Witness Script, Leaf Script, Redeem Script (only non-segwit P2SH), and Bare Output Script (bare outputs, Pay to Multisig, P2PK, P2PKH).
+
+; Control Block
+: [TC] The proof of existence for a Leaf Script.  Appears as a Witness Item in Witness Stacks for P2TR script path inputs.  Only consists of the Inner Key for Script Trees with a single Leaf Script, otherwise provides the Inner Key as well as a Taproot Branch in the form of the hashing partners for the used Leaf Script.
```

Comment:
I second this, and don't know the best way to solve it. To me, the TCs are the easiest to learn because they are fields I am familiar with. So in a way, I want to read all those first. But I was also thinking for the BIP, the raw technical documentation, it probably makes sense to have these all in alphabetical order.

There maybe needs to be another layer of learning material, perhaps supplemental to the BIP. One where the user is taken through a transaction, field by field, and the terms are explained in an order that doesn't have any forward references.

Reply:
Thanks, I’m leaving this unresolved to ponder it a bit more. @satsie, did you see the “Anatomy of a serialized transaction” and table below before you wrote this? Would either of those qualify? Would it work better if the order were different?

Reply:
I just got through everything and that Anatomy section is fantastic!!!! I'm not sure if it makes sense to put it earlier, it might... show the reader the terms and how they are used in a transaction first, then do a deep dive on the terms. Not totally sure. The Anatomy section really helped me understand what a SA is though, something that is mentioned quite early in the doc. So I wonder if general discussion on transaction serialization could be useful to do earlier than later. Sorry I don't have a better answer for this other than "maybe" :sweat_smile: